include/openssl/evp.h

```diff
@@ -1696,6 +1696,8 @@ const char *EVP_PKEY_get0_type_name(const EVP_PKEY *key);
 # define EVP_PKEY_OP_DERIVE              (1<<11)
 # define EVP_PKEY_OP_ENCAPSULATE         (1<<12)
 # define EVP_PKEY_OP_DECAPSULATE         (1<<13)
+# define EVP_PKEY_OP_AUTH_ENCAPSULATE    (1<<14)
+# define EVP_PKEY_OP_AUTH_DECAPSULATE    (1<<15)
```

Comment:
Drop these.  Instead, there should be checks in libcrypto code along the lines of this:

``` C
        if (provauthkey == NULL && kem->decapsulate_init != NULL) {
            ret = kem->decapsulate_init(ctx->op.encap.algctx, provkey, params);
        } else if (provauthkey != NULL && kem->auth_encapsulate_init != NULL) {
            ret = kem->auth_encapsulate_init(ctx->op.encap.algctx, provkey,
                                             provauthkey, params);
       } else {
            ERR_raise(ERR_LIB_EVP, EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
            ret = -2;
            goto err;
        }
```

Reply:
Thanks @levitte